core/src/test/java/de/danoeh/antennapod/core/feed/LocalFeedUpdaterTest.java

```diff
@@ -0,0 +1,258 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import androidx.documentfile.provider.DocumentFile;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.shadows.ShadowMediaMetadataRetriever;
+import org.robolectric.shadows.util.DataSource;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import de.danoeh.antennapod.core.ApplicationCallbacks;
+import de.danoeh.antennapod.core.ClientConfig;
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.robolectric.Shadows.shadowOf;
+
+/**
+ * Test local feeds handling in class LocalFeedUpdater.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class LocalFeedUpdaterTest {
+
+    /**
+     * URL to locate the local feed media files on the external storage (SD card).
+     * The exact URL doesn't matter here as access to external storage is mocked
+     * (seems not to be supported by Robolectric).
+     */
+    private static final String FEED_URL =
+            "content://com.android.externalstorage.documents/tree/primary%3ADownload%2Flocal-feed";
+
+    private Context context;
+    private File localFeedDir1;
+    private File localFeedDir2;
+
+    @Before
+    public void setUp() {
+
+        // Initialize environment
+        context = InstrumentationRegistry.getInstrumentation().getContext();
+        UserPreferences.init(context);
+
+        // Initialize database
+        PodDBAdapter.init(context);
+        PodDBAdapter.deleteDatabase();
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.close();
+
+        // Emulate turned off gpodder.net support in SyncService
```

Comment:
Does this need to be emulated manually? I think the default value of gpodder should already be set to "off".

Reply:
The comment is misleading somehow. The core point is that `ClientConfig.applicationCallbacks` has to be initialized manually to make `GpodnetPreferences.loggedIn()` return false, otherwise I get a `NullPointerException` in `GpodnetPreferences`.

Reply:
Ah okay. How about calling `ClientConfig.initialize`? That should deal with initializing PodDbAdapter, user preferences, etc.

Reply:
Hmm, `ClientConfig.initialize()` doesn't set `ClientConfig.applicationCallbacks`, so I get a `NullPointerException` in `GpodnetPreferences`. The only code that currently sets `ClientConfig.applicationCallbacks` is in the class `ClientConfigurator` that belongs to the `app` module. Did I miss something?

Reply:
Oh my, those callback methods regularly annoy me :D They are a leftover from the time when AntennaPod could be easily white-labelled.

Would it be possible to only mock `applicationCallbacks` but not shared preferences?

Reply:
Yes, I could simplify the test code. Now it only mocks the class `ApplicationCallbacks` but not the classes `Application` and `SharedPreferences`.